cmd/zpool/zpool_main.c

```diff
@@ -9477,8 +9477,16 @@ get_history_one(zpool_handle_t *zhp, void *data)
 	(void) printf(gettext("History for '%s':\n"), zpool_get_name(zhp));
 
 	while (!eof) {
-		if ((ret = zpool_get_history(zhp, &nvhis, &off, &eof)) != 0)
+		if ((ret = zpool_get_history(zhp, &nvhis, &off, &eof)) != 0) {
+			if (ret > 0) {
+				/* ret will be errno when the error message
+				 * didn't get printed by libzfs itself. */
```

Comment:
Instead of treating this as a special case, I'd suggest adding the missing `zpool_standard_error_fmt()` call for the `zpool_history_unpack()` error case in `zpool_get_history()`.   You can also switch the `malloc()` calls in `zpool_get_history()` to `zfs_alloc()`.  This helper can't fail so you can eliminate those ENOMEM error paths on NULL as was done elsewhere in libzfs.

I also happened to notice that the `err` variable really should be initialized in `zpool_get_history()`.  Otherwise there's a case where it may be undefined if the `while` loop is never taken.